tests/WE2E/run_WE2E_tests.py

```diff
@@ -175,8 +175,8 @@ def run_we2e_tests(homedir, args) -> None:
 
         if 'verification' in test_cfg:
             logging.debug(test_cfg['verification'])
```

Comment:
@mkavulich As things stand now, a "verification" section does not have to be included in the test configuration file in order to run a test with verification, but such a section has to exist in the final `config.yaml` file that the generate script reads in because it contains the variable `VX_FCST_INPUT_BASEDIR`, which must be specified for verification (and the call to `check_task_verification` sets this variable).  Your change will still work with all the current WE2E vx tests because they all contain a "verification" section, but if that section is removed from any of the tests because it contains only optional settings (e.g. in the test `MET_verification_only_vx` the section only sets `VX_FCST_MODEL_NAME`, which is optional), the test will fail (I just ran the test both ways myself).  That's why I originally didn't include the call to `check_task_verification` under the if-statement.  This situation may change with upcoming vx PRs, but for now it is safer to always call `check_task_verification`.

Reply:
@gsketefian Thanks for the further info, but I am a bit confused at how this new capability is implemented. Is there no flag for running "verification only"? It appears as if this flag is needed: The reason I made this change is because on Cheyenne (and I presume other non-Hera machines) the `deactivate_tasks` test is now failing due to this path not being set for those platforms.

Reply:
@mkavulich Currently, there is no single flag that turns on/off vx tasks.  Instead, there are the four flags `RUN_TASK_VX_GRIDSTAT`, `RUN_TASK_VX_POINTSTAT`, `RUN_TASK_VX_ENSGRID`, and  `RUN_TASK_VX_ENSPOINT`.  The ultimate goal is to have one flag for deterministic verification and another for ensemble verification.  But I'm a couple of PRs from getting to that point.

The staged forecasts for the vx tasks are only available on Hera.  If I understand you correctly, the `deactivate_tasks` test is failing because `TEST_VX_FCST_INPUT_BASEDIR` is not defined in the machine files of platforms other than Hera.  I think the easy solution is to put those in all the machine files and set it to a blank string for now.  I forgot to do that in one of my PRs.  Could you try that and see if it solves the issue with `deactivate_tasks`?  Thanks.

Reply:
@gsketefian I think you are missing my point: this "TEST_VX_FCST_INPUT_BASEDIR" needs to be set only in the case where the user is doing a VX-only run, correct? So shouldn't there be a flag for VX-only runs? Because VX-only requires a completely different process for setting up directories in the VX tasks, unless I'm mistaken.

Reply:
@mkavulich Yes, `TEST_VX_FCST_INPUT_BASEDIR` needs to be set only when doing a test that runs only vx.  Testing for that requires obtaining values of other flags, specifically `RUN_TASK_RUN_FCST`, `WRITE_DOPOST`, and `RUN_TASK_RUN_POST`.  That's because the check for whether a vx-only run is to see whether UPP is going to be run (either inline or separately) because the forecast inputs for vx tasks are the output files from UPP.  If so, then it's not a vx-only experiment.  Otherwise, it is.  For example, in `config_defaults.yaml`, `VX_FCST_INPUT_BASEDIR` is set like this:
```
VX_FCST_INPUT_BASEDIR: '{{ workflow.EXPTDIR if ((workflow_switches.RUN_TASK_RUN_FCST and task_run_fcst.WRITE_DOPOST) or workflow_switches.RUN_TASK_RUN_POST) }}'
```
We could implement such a check in `run_WE2E_tests.py`, but a lot more vx changes are coming in the next few vx PRs, I decided it's much easier to always set `VX_FCST_INPUT_BASEDIR` by calling `check_task_verification` outside this if-statement.  That in turn requires `TEST_VX_FCST_INPUT_BASEDIR` to at least be defined in the machine files (it should be ok to set it to an empty string), which is something I forgot about in my PR.

I guess we could create a new derived variable/flag for vx-only runs, but that seems redundant since the info can be obtained from `RUN_TASK_RUN_FCST`, `WRITE_DOPOST`, and `RUN_TASK_RUN_POST`.  It would be fine as an internal local variable in `run_WE2E_tests.py` if you want to implement the checks above for when `TEST_VX_FCST_INPUT_BASEDIR` is really necessary, but we should probably just define `TEST_VX_FCST_INPUT_BASEDIR` in all the machine files anyway for completeness (setting it to null strings on platforms on which the forecast files are not staged).

Reply:
Another easy fix for now is to set `TEST_VX_FCST_INPUT_BASEDIR` to a null string if it's not defined in the machine file (instead of erroring out).

Reply:
Maybe it's best to take this conversation over to your latest PR; I'll remove my change from this one.